EIPS/eip-4671.md

```diff
@@ -0,0 +1,291 @@
+    /// @param operator Address allowed to mint a token
+    /// @param owner Address for whom `operator` is allowed to mint a token
+    function delegate(address operator, address owner) external;
+
+    /// @notice Grant one-time minting right to a list of `operators` for a corresponding list of `owners`
+    /// An allowed operator can call the function to transfer rights.
+    /// @param operators Addresses allowed to mint
+    /// @param owners Addresses for whom `operators` are allowed to mint a token
+    function delegateBatch(address[] memory operators, address[] memory owners) external;
+
+    /// @notice Mint a token. Caller must have the right to mint for the owner.
+    /// @param owner Address for whom the token is minted
+    function mint(address owner) external;
+
+    /// @notice Mint tokens to multiple addresses. Caller must have the right to mint for all owners.
+    /// @param owners Addresses for whom the tokens are minted
+    function mintBatch(address[] memory owners) external;
+
+    /// @notice Get the issuer of a token
+    /// @param tokenId Identifier of the token
+    /// @return Address who minted `tokenId`
+    function issuerOf(uint256 tokenId) external view returns (address);
+}
+```
+
+##### Consensus
+
+An interface allowing minting/invalidation of tokens based on a consensus of a predefined set of addresses.
+
+```solidity
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./IERC4671.sol";
+
+interface IERC4671Consensus is IERC4671 {
+    /// @notice Get voters addresses for this consensus contract
+    /// @return Addresses of the voters
+    function voters() external view returns (address[] memory);
+
+    /// @notice Cast a vote to mint a token for a specific address
+    /// @param owner Address for whom to mint the token
+    function approveMint(address owner) external;
+
+    /// @notice Cast a vote to invalidate a specific token
+    /// @param tokenId Identifier of the token to invalidate
+    function approveInvalidate(uint256 tokenId) external;
+}
+```
+
+##### Pull
+
+An interface allowing a token owner to pull his token to a another of his wallets (here `recipient`). The caller must provide a signature of the tuple `(tokenId, owner, recipient)` using the `owner` wallet.
+
+```solidity
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./IERC4671.sol";
+
+interface IERC4671Pull is IERC4671 {
+    // Event emitted when a IERC4671 token is transfered from owner to recipient
+    event Pulled(address owner, address recipient);
+
+    /// @notice Pull a token from the owner wallet to the caller's wallet
+    /// @param tokenId Identifier of the token to transfer
+    /// @param owner Address that owns tokenId
+    /// @param signature Signed data (tokenId, owner, recipient) by the owner of the token
+    function pull(uint256 tokenId, address owner, bytes memory signature) external;
+}
+```
```

Comment:
As suggested when we last talked, I have added an extension that allows self-transfer of the token.

It's a "pull" mechanism, where you have to provide a signed message by the wallet that owns the token in order to transfer it to the caller (which would imply that the caller has access to the current owner's wallet).